### DISCUSSION TITLE
Migration to new API

### DISCUSSION THREAD

**Discussion by Andarius** [software-mansion/react-native-gesture-handler#1867] [Q&A]
I'm trying to migrate all my components to the new API (so I don't have to disable the logs).

So far, I migrated my `PanGestureHandler` and `PinchGestureHandler`.
Now I'm trying to migrate `LongPressGestureHandler`.

What is the equivalent of `onHandlerStateChange` ? `Gesture.LongPress().onStart` seems to do the job but I wanted to make sure it's the right replacement.

Also, `RectButton` triggers the warning. What is the right replacement for it ? 

Thanks !

**Comment by gabimoncha** [comment]
Hi, I'm looking into the same thing but don't know what old api should the new components replace?
@kmagiera @jakub-gonet @j-piasecki @osdnk is there a migration guide somewhere on the roadmap or some good resources to it?

**Reply by j-piasecki** [reply]
We've recently added a migration guide here: https://docs.swmansion.com/react-native-gesture-handler/docs/guides/upgrading-to-2

**Reply by gabimoncha** [reply]
Thank you!
